src/components/pages/CompanyRoleUpdate/index.tsx

```diff
@@ -96,6 +96,14 @@ export default function CompanyRoles() {
     setSelectedRoles(rolesArr)
   }
 
+  const checkDisabledSubmit = () => {
+    const roles = preSelectedRoles.map((preRole) => preRole.companyRoles)
+    return (
+      JSON.stringify(roles) === JSON.stringify(selectedRoles) ||
```

Comment:
This is unreliable because it would fail with
  `["roleA","roleB"] === ["roleB","roleA"]`
we must sort the array before but then what about if for some reason we have
  `["roleA","roleB"] === ["roleA","roleB","roleB"]`
maybe we know from the rest of the code that this cases can't happen but that is very risky and not robust because if the implementation changes nobody will check here and then it fails.

The entire logic in both files CompanyRoleUpdate and UpdateCompanyRole seems flawed because it uses Arrays for a case where using a Set would be more appropriate and much easier to implement. Will file a ticket so we can improve the implementation.

Reply:
@oyo  I changed the logic. Please review now